selfupdate/detect.go

```diff
@@ -49,7 +55,16 @@ func findAssetFromReleasse(rel *github.RepositoryRelease, suffixes []string, tar
 	for _, asset := range rel.Assets {
 		name := asset.GetName()
 		for _, s := range suffixes {
-			if strings.HasSuffix(name, s) {
+			if strings.HasSuffix(name, s) { // require version, arch etc
+				if len(settings.filters) > 0 {
+					for _, filter := range settings.filters { // extra condition to select among several release artifacts
+						if filter.MatchString(name) {
+							return &asset, ver, true
+						}
+					}
+					break
```

Comment:
It would be better to put `if len(settings.fiters) > 0 { ... }` code just after `name :=`. I prefer an early return to avoid deep nesting blocks. And please add log output for making debug easier.

```go
name := asset.GetName()
if len(settings.filters) > 0 {
    matched := false
    for _, filter := range settings.filters { // extra condition to select among several release artifacts
        if filter.MatchString(name) {
            matched = true
            break
        }
    }
    if !matched {
        log.Println("Skip", name, "since any filter did not match")
        return nil, false
    }
}
```

Reply:
ok, makes sense